api/types/installers/installer.sh.tmpl

```diff
@@ -1,4 +1,10 @@
 #!/bin/sh
+
```

Comment:
The lack of `set -eu` makes me sad.

```suggestion
set -eu
```

Don't take this suggestion lightly, though, the script needs to be thoroughly retested if you add this.

I'd also strongly recommend adding the body of the script to a `main` function and scoping variables locally (if a shebang change to bash is OK), as a starter. I realize some of this is pre-existing, but I also didn't get a chance to talk about it before. :)

Reply:
Are we sure that the current version of the script will work after `set -eu` is added? Maybe this is a silly question, but nothing is really this script from "failing successfully" now on EC2.

Reply:
@codingllama I understand your concerns, and I agree that we should not ignore any execution failures. My point is more about the scope of this PR. This PR doesn't decrease the quality of the script (`set -eu` was missing before the changes and is missing after).
I created a separate issue to trace this problem https://github.com/gravitational/teleport/issues/22199

Reply:
Thanks for raising the issue.